controllers/nnf_workflow_controller_helpers.go

```diff
@@ -1442,14 +1503,31 @@ func (r *NnfWorkflowReconciler) getMPIJobList(ctx context.Context, workflow *dws
 	return jobList, nil
 }
 
+func (r *NnfWorkflowReconciler) getMPIJobs(ctx context.Context, workflow *dwsv1alpha2.Workflow, index int) (*mpiv2beta1.MPIJobList, error) {
+	// Get the MPIJobs for this workflow and directive index
+	matchLabels := dwsv1alpha2.MatchingWorkflow(workflow)
+	if index >= 0 {
+		matchLabels[nnfv1alpha1.DirectiveIndexLabel] = strconv.Itoa(index)
+	}
+
+	jobList := &mpiv2beta1.MPIJobList{}
+	if err := r.List(ctx, jobList, matchLabels); err != nil {
+		return nil, nnfv1alpha1.NewWorkflowError("could not retrieve MPIJobs").WithError(err)
+	}
+
+	return jobList, nil
+}
+
 func (r *NnfWorkflowReconciler) getContainerJobs(ctx context.Context, workflow *dwsv1alpha2.Workflow, index int) (*batchv1.JobList, error) {
 	// Get the jobs for this workflow and directive index
 	matchLabels := dwsv1alpha2.MatchingWorkflow(workflow)
-	matchLabels[nnfv1alpha1.DirectiveIndexLabel] = strconv.Itoa(index)
+	if index >= 0 {
```

Comment:
What would cause it to be less than zero?  (I'm still digging...)

Reply:
I made this and `getMPIJobs` have the ability to use `-1` for `index` for when you don't want to filter by index. `deleteContainers` is called with an index of `-1` when the workflow is deleted (rather than a proper Teardown->delete). See here: https://github.com/NearNodeFlash/nnf-sos/blob/da071ea41ed1194601ca43901f9dbb474837736d/controllers/nnf_workflow_controller.go#L128